src/Raven.Server/Utils/RavenCli.cs

```diff
@@ -0,0 +1,706 @@
+        {
+            public Command Command;
+            public ConcatAction PrevConcatAction;
+            public List<string> Args;
+        }
+
+        private class ParsedLine
+        {
+            public LineState LineState;
+            public string ErrorMsg;
+            public List<ParsedCommand> ParsedCommands = new List<ParsedCommand>();
+        }
+
+
+        public static bool Start(RavenServer server)
+        {
+            _server = server;
+
+            try
+            {
+                return StartCli();
+            }
+            catch (Exception ex)
+            {
+                // incase of cli failure - prevent server from going down, and switch to a (very) simple fallback cli
+                Console.WriteLine("\nERROR in CLI:" + ex);
+                Console.WriteLine("\n\nSwitching to simple cli...");
+
+                while (true)
+                {
+                    Console.Write("(simple cli)>");
+                    Console.Out.Flush();
+                    var line = Console.ReadLine();
+                    if (line == null)
+                        continue;
+                    switch (line)
+                    {
+                        case "quit":
+                        case "q":
+                            return false;
+                        case "reset":
+                            return true;
+                        case "log":
+                            LoggingSource.Instance.EnableConsoleLogging();
+                            LoggingSource.Instance.SetupLogMode(LogMode.Information, Path.Combine(AppContext.BaseDirectory, _server.Configuration.Logs.Path));
+                            break;
+                        case "logoff":
+                            LoggingSource.Instance.DisableConsoleLogging();
+                            LoggingSource.Instance.SetupLogMode(LogMode.None, Path.Combine(AppContext.BaseDirectory, _server.Configuration.Logs.Path));
+                            break;
+                        case "h":
+                        case "help":
+                            Console.WriteLine("Available commands: quit, reset, log, logoff");
+                            Console.Out.Flush();
+                            break;
+                    }
+                }
+            }
+        }
+
+        public static bool StartCli()
+        {
+            var ctrlCPressed = false;
+            Console.CancelKeyPress += (sender, args) =>
+            {
+                ctrlCPressed = true;
+            };
+
+            while (true)
+            {
+                PrintCliHeader();
+                var line = Console.ReadLine();
+                Console.Out.Flush();
+
+                if (line == null)
+                {
+                    Thread.Sleep(75); //waiting for Ctrl+C 
+                    if (ctrlCPressed)
+                        break;
+                    Console.WriteLine("End of standard input detected, switching to server mode...");
```

Comment:
on unix daemon, in docker, and also in windows nano server or any other console not supporting ReadKey